system/jlib/jbuff.cpp

```diff
@@ -243,13 +249,21 @@ void MemoryBuffer::_realloc(size32_t newLen)
                 newMax = FIRST_CHUNK_SIZE;
             while (newLen > newMax)
             {
-                newMax += newMax;
+                size32_t newMaxTmp = newMax + newMax;
+                if (newMaxTmp < newMax)
+                    RaiseOutOfMemException(-11, newMaxTmp, newMax);
```

Comment:
minor: the code would be simpler, if used an inline func. to check, throw and/or return new value